tiledb/sm/array_schema/array_schema.cc

```diff
@@ -545,95 +586,150 @@ Status ArraySchema::drop_attribute(const std::string& attr_name) {
   return Status::Ok();
 }
 
-Status ArraySchema::deserialize(ConstBuffer* buff) {
+shared_ptr<ArraySchema> ArraySchema::deserialize(
+    ConstBuffer* buff, const URI& uri) {
+  Status st;
```

Comment:
Throughout this function, we've not been doing any data validation. It's time to add it in. Exceptions for these should _not_ be combined with the ones for buffer-read errors. The buffer I/O will be going away at some point, but the validation will not.

I'm agnostic about writing separate validation functions vs. just coding them into this function. We'll want them extracted at some point, but I don't see need to do it at the moment. Partly that's because we might an error in validation either to return false or to throw, and it's not clear now which is better.

I've not listed all the places we should be validating.